test/OpenTelemetry.Contrib.Shared.Tests/SqlProcessorTestCases.cs

```diff
@@ -15,36 +18,68 @@ public static class SqlProcessorTestCases
         Converters = { new JsonStringEnumConverter() },
     };
 
-    public static IEnumerable<object[]> GetTestCases()
+    private static readonly Uri Uri = new("https://raw.githubusercontent.com/open-telemetry/semantic-conventions/refs/heads/main/docs/non-normative/database-test-cases/db-sql-test-cases.json");
+    private static readonly HashSet<string> DbSystemTestCasesToExecute = ["other_sql"];
+
+    public static IEnumerable<object[]> GetSemanticConventionsTestCases()
```

Comment:
If you use `TheoryData<>` we can make things strongly typed.

Reply:
It's a bit more complex as you can't use the `yield` keyword in that case.

Reply:
OK, I can look to refactor it separately.

I don't think the tests are performance sensitive enough to need to worry about using `yield` (rather than just return them all) and I think it would be better to enjoy the type safety if it's a choice between the two.

Reply:
I can update

Reply:
I've updated to `TheoryData`. For now I preferred using the `TestCase` type directly. This results in nicer display in VS for example:

<img width="589" height="134" alt="image" src="https://github.com/user-attachments/assets/db0d3281-3bf9-4f45-9eb6-138d675e2cc8" />

The downside is needing to implement `IXunitSerializable`. The alternative is passing back `string` instances instead, which works but is less pleasant in the UI.

Reply:
I think that's good enough as it lets you easily find/see the specific test cases.